navigation/navigation-common/src/main/java/androidx/navigation/NavDestination.kt

```diff
@@ -312,6 +312,14 @@ public open class NavDestination(
      * @see NavController.navigate
      */
     public fun addDeepLink(navDeepLink: NavDeepLink) {
+        val missingRequiredArguments = arguments.filterValues { !it.isDefaultValuePresent }
+            .keys
+            .filter { it !in navDeepLink.arguments }
+        require(missingRequiredArguments.isEmpty()) {
+            "Deep link ${navDeepLink.uriPattern} can't be used to open destination $this.\n" +
+                "Following required arguments are missing: $missingRequiredArguments"
+        }
```

Comment:
It would be great to have a check like this to make sure deep links will be able to open this destination. But I'm not sure I've selected the right place for this. We can add the required parameters after deepLink is added to bypass this validation.

Reply:
Maybe also add validation after a new argument was added?

Reply:
Ok. I'll fix it

Reply:
@jbw0033, I've fixed tests in the latest commit (https://github.com/androidx/androidx/pull/204/commits/d1cced081c73c3b895bec1605f92e96b6b3f616e), can you take a look if I did it right?